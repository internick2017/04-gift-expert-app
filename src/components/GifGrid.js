import React, { useState, useEffect } from 'react'

export const GifGrid = ({ category }) => {

    useEffect(() => {
        getGifs();
    }, [])
    const getGifs = async () => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=Bleach&limit=10&api_key=dOv0LM1e9OjbMAvId5PDvdTySsgzD3ri';
        const resp = await fetch(url);
        const { data } = await resp.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })

        console.log(gifs)
    }

    return (
        <>
            <h3>{category}</h3>
            <button></button>
        </>
    )
}
