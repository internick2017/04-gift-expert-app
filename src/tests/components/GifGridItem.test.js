import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('pruebas en gif grid item', () => {
    const title = 'un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('debe mostrar componente correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el title ', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('debe de tener la imagen  igual al url y al de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('deve tener la clase animate__rubberBand', () => {
        const div = wrapper.find('.card')
        const className = div.props().className;
        expect(className.includes('animate__rubberBand')).toBe(true);
    })
    



})
