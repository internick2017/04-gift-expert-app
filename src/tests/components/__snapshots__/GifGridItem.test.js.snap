// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en gif grid item debe mostrar componente correctamente  1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    un titulo
  </p>
</div>
`;
